### ISSUE TITLE
Allow 're-assignment' of pores when creating Geometry and Physics?

### ISSUE THREAD

**Issue by jgostick** [PMEAL/OpenPNM#1407]
at the moment, if you assign all 100 pores to a geometry, then try to create a second geometry, we throw an error.  Maybe it would be convenient to allow this.  For instance, if I'm importing a network, then I use the `Imported` geometry class, all the pores are assigned to it.  But what if I subsequently want to create a boundary geometry and assign pores 1, 2 and 3?  It may be handy to allow this:

``` python
geo1 = openpnm.geometry.Imported(network=pn)
geo2 = openpnm.geometry.Boundary(network=pn, pores=[1, 2, 3])
```

Interestingly, we used to offer `set_locations` and `remove_locations` methods, but have removed them.  So there is not actually a way to do this.  It used to work like this:

``` python
geo1 = openpnm.geometry.Imported(network=pn)
geo2 = openpnm.geometry.Boundary(network=pn, pores=None, throats=None)
geo1.remove_locations(pores=[1, 2, 3])
geo2.set_locations(pores=[1, 2, 3])
```

**Comment by jgostick** [comment]
Now that I'm playing with it a bit, here is a proposal:

This should be handled by the `Project`.  We should add a function like `reassign_locations`, that shifts all info from one object to another, including pore scale models, properties, labels, etc. Like this:

``` python
proj.reassign_locations(source=geo1, target=geo1, pores=[1, 2, 3], throats=[0, 5, 8], mode='all')
```

`mode` could also be a list like ['labels', 'props', 'models']

**Comment by jgostick** [comment]
This was addressed in PR #1705
